### PR TITLE
[pipeline] refactor custom pipeline init

### PR DESCRIPTION
**Pull request by star-e**
changed `effectSettings` loading procedure.
add more constness to gfx.
updated native pipeline init/destroy procedure.
updated native custom pipeline
refactored native Pass. The logic is the same as web.

### Changelog

* changed game.ts
* fixed `validPunctualLights` jsb getter/setter bug.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/core/ProgramLib.cpp [hunk 1]
```diff
@@ -178,15 +178,7 @@ void ProgramLib::registerEffect(EffectAsset *effect) {
...
-
-    for (auto &tech : effect->_techniques) {
-        for (auto &pass : tech.passes) {
-            // grab default property declaration if there is none
-            if (pass.propertyIndex.has_value() && !pass.properties.has_value()) {
-                pass.properties = tech.passes[pass.propertyIndex.value()].properties;
-            }
-        }
-    }
+    render::addEffectDefaultProperties(*effect);
```
Comment (star-e) [inline native/cocos/renderer/core/ProgramLib.cpp:181]: moved function to `render::addEffectDefaultProperties`

File: native/cocos/renderer/pipeline/PipelineSceneData.h [hunk 2]
```diff
@@ -64,6 +64,7 @@ class CC_DLL PipelineSceneData : public RefCounted {
     inline const RenderObjectList &getRenderObjects() const { return _renderObjects; }
     inline void setRenderObjects(RenderObjectList &&ro) { _renderObjects = std::forward<RenderObjectList>(ro); }
     inline const ccstd::vector<const scene::Light *> &getValidPunctualLights() const { return _validPunctualLights; }
+    inline void setValidPunctualLights(ccstd::vector<const scene::Light *> lights) { _validPunctualLights = std::move(lights); }
```
Comment (star-e) [inline native/cocos/renderer/pipeline/PipelineSceneData.h:67]: `validPunctualLights` is accessed with getter/setter in ts.

**Comment by star-e** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Review by pandamicro — APPROVED** [review]

**Pull request merged**